deploy/helm/secret-operator/templates/secret_migration_job.yaml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "operator.fullname" . }}-secret-migration
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+  labels:
+    {{- include "operator.labels" . | nindent 4 }}
+spec:
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "operator.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "operator.fullname" . }}-serviceaccount
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+      - name: migrate-secret
+        image: "{{ .Values.secretMigrationJob.image.repository }}:1.0.0-stackable{{ .Chart.AppVersion }}"
+        imagePullPolicy: {{ .Values.secretMigrationJob.image.pullPolicy }}
+        resources:
+            {{ .Values.secretMigrationJob.resources | toYaml | nindent 12 }}
+        command: ["bash", "-c"]
+        args:
+        - |
+          set -uo pipefail
+          SOURCE_NAMESPACE=default
+          TARGET_NAMESPACE={{ .Values.secretClasses.tls.caSecretNamespace | default .Release.Namespace }}
+
+          kubectl get secret -n $SOURCE_NAMESPACE secret-provisioner-tls-ca
+          exit_status=$?
+          # only continue if secret exists
+          if [ $exit_status -eq 0 ]; then
+            echo "secret exists in namespace $SOURCE_NAMESPACE"
+            kubectl get secret -n $TARGET_NAMESPACE secret-provisioner-tls-ca
+            exit_status=$?
+            # only continue if secret in target namespace does NOT exist
+            if [ $exit_status -eq 1 ]; then
+              echo "secret does not exist in namespace $TARGET_NAMESPACE"
+              # copy secret from default to {{ .Values.secretClasses.tls.caSecretNamespace | default .Release.Namespace }}
+              kubectl get secret -n $SOURCE_NAMESPACE secret-provisioner-tls-ca -o json | jq 'del(.metadata["namespace","creationTimestamp","resourceVersion","selfLink","uid"])' | kubectl apply -n $TARGET_NAMESPACE -f -
+            fi
+          fi
```

Comment:
Run shellcheck please

Reply:
Done, it had a smaller complaint about something that was irrelevant in that case but I fixed it anyways.